### PR TITLE
Issue #2585019: Implementing doExecute() for the DataListItemRemove a…

### PR DESCRIPTION
**Pull request by emclaughlin1215**
…ction.

See d.o issue: https://www.drupal.org/node/2585019 and https://www.drupal.org/node/2572119

### CODE REVIEW

File: src/Plugin/RulesAction/DataListItemRemove.php [hunk 1]
```diff
@@ -33,12 +33,14 @@
 class DataListItemRemove extends RulesActionBase {
 
   /**
-   * {@inheritdoc}
+   * Executes the Plugin.
```
Comment (klausi) [inline src/Plugin/RulesAction/DataListItemRemove.php]: This should be a short description what the action does.

**Comment by emclaughlin1215** [comment]
Addressed the comments and pushed up changes.

File: src/Plugin/RulesAction/DataListItemRemove.php [hunk 2]
```diff
@@ -33,12 +33,14 @@
...
+   * @param array $list
+   *   An array to remove an item from.
+   * @param mixed $item
+   *   An item to remove from the array.
    */
-  public function execute() {
-    $list = $this->getContextValue('list');
-    $item = $this->getContextValue('item');
-
+  public function doExecute($list, $item) {
```
Comment (klausi) [inline src/Plugin/RulesAction/DataListItemRemove.php:43]: should be protected, not public.

**Comment by klausi** [comment]
continued in #300

**Pull request closed without merging**